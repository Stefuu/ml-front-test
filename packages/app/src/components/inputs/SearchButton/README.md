# Search Button

## Default Search Button