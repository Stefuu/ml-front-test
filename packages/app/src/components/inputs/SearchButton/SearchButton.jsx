import tag from 'clean-tag'
import styled from 'styled-components'
import { theme } from 'styled-tools'

const SearchButton = styled(tag.button)`
    background-color: ${theme('colors.b5')}
    cursor: pointer;
    width: 42px;
    height: 100%;
    display: flex;
    justify-content: center;
    outline: none;
`

export default SearchButton
