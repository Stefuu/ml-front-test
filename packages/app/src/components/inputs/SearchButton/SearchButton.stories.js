import React from 'react'
import { storiesOf } from '@storybook/react'
import { doc } from 'storybook-readme'
import readmeFile from './README.md'
import centered from '@storybook/addon-centered'
import theme from '../../themes/default'
import { ThemeProvider } from 'styled-components'
import SearchButton from './SearchButton'
import GlobalStyle from '../../modules/GlobalStyle'
import RetinaImage from '../../behaviour/RetinaImage'
import SearchImageRegular from '../../../assets/ic_Search.png'
import SearchImageRetina from '../../../assets/ic_Search@2x.png'
import Flex from '../../elements/Flex'

storiesOf('app/components/inputs/SearchButton', module)
  .addParameters({ jest: ['InputText.test.js'] })
  .add('README', doc(readmeFile))
  .addDecorator(centered)
  .addParameters({ jest: ['SearchButton.test.js'] })
  .add('simple', () => (
    <ThemeProvider theme={theme}>
      <React.Fragment>
        <GlobalStyle />
        <SearchButton>
          <Flex
            alignItems='center'
            width={17}
            height={36}>
            <RetinaImage
              width={20}
              height={18}
              regular={SearchImageRegular}
              retina={SearchImageRetina}
              alt='Logo' />
          </Flex>
        </SearchButton>
      </React.Fragment>
    </ThemeProvider>
  ))
