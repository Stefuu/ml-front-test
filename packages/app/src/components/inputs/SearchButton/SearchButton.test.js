import React from 'react'
import { render } from 'react-testing-library'
import SearchButton from './SearchButton'

describe('SearchButton', () => {
  it('Should render [snapshot]', async () => {
    const { container } = render(<SearchButton />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
