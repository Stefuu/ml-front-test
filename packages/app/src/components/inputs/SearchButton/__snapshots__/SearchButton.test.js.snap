// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton Should render [snapshot] 1`] = `
<button
  class="sc-bdVaJa HHisZ"
  data-testid="search-button"
/>
`;
